lib/ParseSIL/ParseSIL.cpp

```diff
@@ -885,6 +887,158 @@ void SILParser::convertRequirements(SILFunction *F,
   }
 }
 
+// SWIFT_ENABLE_TENSORFLOW
+/// Parse a SIL constant value (one of the following categories):
+/// - An integer datatype and literal (i32 42).
+/// - A floating point datatype and literal (f64 3.14).
+///   - The literal value may be in either decimal format or the hexadecimal
+///     format used by the 'float_literal' instruction.
+/// - A metatype (the instance type is parsed) ($Float).
+/// - An aggregate ([i32 1, i64 2, f32 3.0]).
```

Comment:
what is the use case for "aggregate" here? also i didn't find its mention in the SIL.rst doc above.

Reply:
Aggregates primarily represent constant arrays (e.g. tensor shapes).
They may also represent other aggregate types like structs or tuples.

Aggregate constants are useful for TensorFlow ops that have an array attribute, such as [`Conv2D`](https://github.com/tensorflow/tensorflow/blob/694b75a24bce416555425dedc58b0cdcd0d52c1e/tensorflow/core/ops/nn_ops.cc#L265):
```c++
REGISTER_OP("Conv2D")
    .Input("input: T")
    .Input("filter: T")
    .Output("output: T")
    .Attr("T: {half, bfloat16, float, double}")
    .Attr("strides: list(int)")
    .Attr("use_cudnn_on_gpu: bool = true")
    .Attr(GetPaddingAttrString())
    .Attr(GetConvnetDataFormatAttrString())
    .Attr("dilations: list(int) = [1, 1, 1, 1]")
    .SetShapeFn(shape_inference::Conv2DShape);
```

I'll add more information about aggregates in the doc comment.